orchagent/qosorch.h

```diff
@@ -1,20 +1,150 @@
-#pragma once
+#ifndef SWSS_QOSORCH_H
+#define SWSS_QOSORCH_H
 
-#include "portsorch.h"
-#include "orch.h"
 #include <map>
+#include "orch.h"
+#include "portsorch.h"
 
 namespace swss {
+
+const std::string delimiter                         = ":";
+const std::string tc_to_queue_field_name            = "tc_to_queue_map";
+const std::string dscp_to_tc_field_name             = "dscp_to_tc_map";
+const std::string scheduler_field_name              = "scheduler";
+const std::string wred_profile_field_name           = "wred_profile";
+const char        ref_start                         = '[';
+const char        ref_end                           = '[';
+const std::string yellow_max_threshold_field_name   = "yellow_max_threshold";
+const std::string green_max_threshold_field_name    = "green_max_threshold";
+
+const std::string scheduler_algo_type_field_name    = "type";
+const std::string scheduler_algo_DWRR               = "DWRR";
+const std::string scheduler_algo_WRR                = "WRR";
+const std::string scheduler_algo_PRIORITY           = "PRIORITY";
+const std::string scheduler_weight_field            = "weight";
+const std::string scheduler_priority_field          = "priority";
+
+typedef std::map<string, sai_object_id_t> qos_object_map;
+typedef std::pair<string, sai_object_id_t> qos_object_map_pair;
+
+typedef std::map<string, qos_object_map*> qos_type_map;
+typedef std::pair<string, qos_object_map*> qos_type_map_pair;
+
+
+class QosMapHandler
+{
+public:
+    bool processWorkItem(Consumer& consumer);
+    virtual bool isValidTable(string &tableName) = 0;
+    virtual bool convertFieldValuesToAttributes(KeyOpFieldsValuesTuple &tuple, std::vector<sai_attribute_t> &attributes) = 0;
+    virtual void freeAttribResources(std::vector<sai_attribute_t> &attributes) = 0;
```

Comment:
this function seems to be the same for all inheritors. It could be implemented here without =0.